index.bs

```diff
@@ -289,12 +341,18 @@ A {{FileSystemFileHandle}}'s associated [=FileSystemHandle/entry=] must be a [=f
 The <dfn method for=FileSystemFileHandle>getFile()</dfn> method steps are:
 
 1. Let |result| be [=a new promise=].
+1. Let |entry| be the the result of [=getting the entry=] for [=this=].
 1. Run these steps [=in parallel=]:
   1. Let |access| be the result of running [=this=]'s [=FileSystemHandle/entry=]'s
      [=file system entry/query access=] given "`read`".
   1. If |access| is not "{{PermissionState/granted}}",
      [=/reject=] |result| with a "{{NotAllowedError}}" {{DOMException}} and abort.
-  1. Let |entry| be [=this=]'s [=FileSystemHandle/entry=].
+
+  1. If |entry| is `null`, [=/reject=] |result| with a
+     "{{NotFoundError}}" {{DOMException}} and abort.
+  1. If |entry| is not a [=file entry=] [=/reject=] |result| with a
+     "{{TypeMismatchError}}" {{DOMException}} and abort.
```

Comment:
I realize (as per https://github.com/whatwg/fs/issues/57#issuecomment-1277022240) that `TypeMismatchError` is deprecated, but it's used elsewhere in this API and is thrown implicitly in a number of places already (e.g. most of the places affected by #11). This PR helps make it more explicit where this error comes from, but obviously it's not great to be throwing a deprecated exception so often... any suggestions on what to do here?

Reply:
I think aligning with @domenic's suggestions is still doable. Typically exceptions are not relied upon.

Reply:
Since "locating an entry" now takes into account a "kind", I've changed this to assert that the "entry" you get back is of the right kind